### PR TITLE
UI/importers: Translate capture sources depending on display server

### LINKED ISSUE #5323: [Bug] If a scene collection exported from Windows is imported into OBS on Linux, OBS crashes with a SIGSEGV in obs_module_get_locale_string if the Game Capture source is clicked at all.
### Operating System Info

Other

### Other OS

Arch Linux

### OBS Studio Version

Other

### OBS Studio Version (Other)

27.0.1-3

### OBS Studio Log URL

https://obsproject.com/logs/52aCr5ZFXILrTbER

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

I expected to be able to open the Game Capture source and have it show "No Properties available," giving me the ability to then remove it and replace it with the Linux equivalent.

### Current Behavior

Left-clicking, right-clicking, doesn't matter, any click on a Game Capture source imported from Windows force-closes the window with a Segmentation Fault logged if run from the terminal. No crash log appears. (or if there is one, I'm not sure where or how I'm supposed to upload it)

### Steps to Reproduce

1. Make a new Scene on the Windows version of OBS, add a Game Capture source.
2. Export the scene collection.
3. Import that same collection on a Linux machine.
4. Attempt to interact with the Game Capture source.

### Anything else we should know?

The version of OBS I am using is from the Arch Linux repos, so chances are this bug may have been found and patched already in Master.

### PR DESCRIPTION
**Pull request by SarenDev**
### Description

Fixes a crash when importing scenes from Windows by translating window and game capture sources to `xcomposite` or `pipewire` depending on the display server used.
This change prevents `xcomposite` from being assigned on non-X11 systems, which causes the crash due to the module not being loaded.

### Motivation and Context

Fixes #5323

### How Has This Been Tested?

Tested under Ubuntu 24.10 with KDE Plasma 6.1.5 under both Wayland and X11

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by SarenDev** [comment]
I also noticed that the Windows and MacOS translation implementations are missing translations for `pipewire`. Should they be added to this request or a new one?

File: UI/importers/studio.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@
 ******************************************************************************/
 
 #include "importers.hpp"
+#include <obs-nix-platform.h>
```
Comment (tytan652) [inline UI/importers/studio.cpp:19]: This header must not be included on Windows and macOS.

File: UI/importers/studio.cpp [hunk 2]
```diff
@@ -109,7 +110,13 @@ void TranslateOSStudio(Json &res)
 		ClearTranslation("av_capture_input", "v4l2_input");
 		ClearTranslation("dshow_input", "v4l2_input");
 
-		ClearTranslation("window_capture", "xcomposite_input");
+		if (obs_get_nix_platform() == OBS_NIX_PLATFORM_X11_EGL) {
+			ClearTranslation("game_capture", "xcomposite_input");
+			ClearTranslation("window_capture", "xcomposite_input");
+		} else {
+			ClearTranslation("game_capture", "pipewire-window-capture-source");
+			ClearTranslation("window_capture", "pipewire-window-capture-source");
```
Comment (tytan652) [inline UI/importers/studio.cpp:118]: Wrong ID in use.

```suggestion
			ClearTranslation("game_capture", "pipewire-screen-capture-source");
			ClearTranslation("window_capture", "pipewire-screen-capture-source");
```

**Review by tytan652 — CHANGES_REQUESTED** [review]
> Translate capture sources depending on WM

This is not a good wording, Wayland and X11 are not WMs but windowing systems.

**Comment by SarenDev** [comment]
> > Translate capture sources depending on WM
> 
> This is not a good wording, Wayland and X11 are not WMs but windowing systems.

Sorry, I realized that after I submitted my request

File: UI/importers/studio.cpp (hunk 1 above)
Comment (SarenDev) [reply UI/importers/studio.cpp:20]: Added a platform check

**Comment by norihiro** [comment]
The commit d7b1be3c1 has a too long description in a line, 88 characters, limit 72:
> Use pipewire-screen-capture instead of the currently discouraged pipewire-window-capture

**Comment by SarenDev** [comment]
> The commit [d7b1be3](https://github.com/obsproject/obs-studio/commit/d7b1be3c1e50a416b2f2ef355b12b94682d82a68) has a too long description in a line, 88 characters, limit 72:
> 
> > Use pipewire-screen-capture instead of the currently discouraged pipewire-window-capture

Corrected, apologies

**Comment by norihiro** [comment]
The folding looks fine. Though, three commits should be squashed into one commit.

**Comment by Lain-B** [comment]
Agreed with norihiro, squash the latter two commits into the first commit and it should be good.

**Review by Lain-B — APPROVED** [review]

**Comment by RytoEX** [comment]
If a user imports a scene collection in Wayland, and then swaps to X11 later, what happens to these sources? Do they simply not load? Will OBS crash if you try to access their properties?

cc @kkartaltepe since we might have had prior conversations about what to do about this.

If this fixes a GitHub issue, use this syntax in the PR description:
> Fixes #5323

**Comment by kkartaltepe** [comment]
> If a user imports a scene collection in Wayland, and then swaps to X11 later, what happens to these sources? Do they simply not load? Will OBS crash if you try to access their properties?

They just have no properties and are not useful. This was considered fine when we decided to split the capture plugins between the platforms. I don't have a problem with the implementation as is.

**Review by RytoEX — APPROVED** [review]

**Pull request merged**